File: phi_intelligence/Phi-TMS/frontend/src/types/project.ts
// Project Status Enum
export enum ProjectStatus {
  PLANNING = 'PLANNING',
  ACTIVE = 'ACTIVE',
  ON_HOLD = 'ON_HOLD',
  COMPLETED = 'COMPLETED',
  CANCELLED = 'CANCELLED',
}

// Priority Enum
export enum Priority {
  LOW = 'LOW',
  MEDIUM = 'MEDIUM',
  HIGH = 'HIGH',
  CRITICAL = 'CRITICAL',
}

// Region Enum
export enum Region {
  UK = 'UK',
  INDIA = 'INDIA',
  BOTH = 'BOTH',
}

// User (simplified from backend)
export interface User {
  id: string;
  email: string;
  firstName: string;
  lastName: string;
  role: string;
  department?: string;
  profilePictureUrl?: string;
}

// Project interface
export interface Project {
  id: string;
  name: string;
  description?: string;
  clientName?: string;
  projectCode: string;
  status: ProjectStatus;
  startDate: string;
  endDate?: string;
  estimatedHours?: number;
  actualHours?: number;
  budget?: number;
  region: Region;
  priority: Priority;
  projectLeadId: string;
  createdById: string;
  createdAt: string;
  updatedAt: string;
  
  // Relations
  projectLead?: User;
  createdBy?: User;
  members?: ProjectMember[];
  _count?: {
    tasks?: number;
    members?: number;
    timeLogs?: number;
  };
}

// Project Member interface
export interface ProjectMember {
  id: string;
  projectId: string;
  userId: string;
  role: string;
  allocationPercentage: number;
  joinedDate: string;
  leftDate?: string;
  hoursAllocated?: number;
  hoursLogged?: number;
  
  // Relations
  user?: User;
  project?: Project;
}

// Project Metrics
export interface ProjectMetrics {
  totalTasks: number;
  completedTasks: number;
  inProgressTasks: number;
  blockedTasks: number;
  overdueTasks: number;
  totalHours: number;
  estimatedHours: number;
  completionPercentage: number;
  teamSize: number;
  budgetUsed?: number;
  budgetRemaining?: number;
}

// Project Timeline
export interface ProjectTimeline {
  projectId: string;
  startDate: string;
  endDate?: string;
  currentDate: string;
  progressPercentage: number;
  daysElapsed: number;
  daysRemaining?: number;
  isOverdue: boolean;
  milestones: Milestone[];
  recentActivity: Activity[];
}

export interface Milestone {
  id: string;
  projectId: string;
  name: string;
  description?: string;
  dueDate: string;
  completedDate?: string;
  status: 'PENDING' | 'IN_PROGRESS' | 'COMPLETED';
  orderIndex: number;
}

export interface Activity {
  id: string;
  type: string;
  description: string;
  userId: string;
  userName: string;
  timestamp: string;
  relatedEntityId?: string;
  relatedEntityType?: string;
}

// Form Data interfaces
export interface ProjectFormData {
  name: string;
  description?: string;
  clientName?: string;
  projectCode: string;
  status?: ProjectStatus;
  startDate: string;
  endDate?: string;
  estimatedHours?: number;
  budget?: number;
  projectLeadId: string;
  region: Region;
  priority: Priority;
}

export interface ProjectMemberFormData {
  userId: string;
  role: string;
  allocationPercentage?: number;
  hoursAllocated?: number;
}

// Filters
export interface ProjectFilters {
  status?: ProjectStatus;
  projectLeadId?: string;
  region?: Region;
  priority?: Priority;
  search?: string;
}

// Status badge colors
export const PROJECT_STATUS_COLORS: Record<ProjectStatus, string> = {
  [ProjectStatus.PLANNING]: 'bg-gray-100 text-gray-800',
  [ProjectStatus.ACTIVE]: 'bg-blue-100 text-blue-800',
  [ProjectStatus.ON_HOLD]: 'bg-yellow-100 text-yellow-800',
  [ProjectStatus.COMPLETED]: 'bg-green-100 text-green-800',
  [ProjectStatus.CANCELLED]: 'bg-red-100 text-red-800',
};

// Priority badge colors
export const PRIORITY_COLORS: Record<Priority, string> = {
  [Priority.LOW]: 'bg-gray-100 text-text-secondary',
  [Priority.MEDIUM]: 'bg-blue-100 text-blue-700',
  [Priority.HIGH]: 'bg-orange-100 text-orange-700',
  [Priority.CRITICAL]: 'bg-red-100 text-red-700',
};

File: phi_intelligence/Phi-TMS/backend/app/schemas/project.py
"""Project Pydantic schemas"""
from typing import Optional
from pydantic import BaseModel, Field
from datetime import datetime

from app.models.enums import ProjectStatus, Priority, Region


class CreateProjectRequest(BaseModel):
    """Create project request schema"""
    name: str = Field(..., min_length=1, max_length=200)
    description: Optional[str] = None
    client_name: Optional[str] = None
    project_code: str = Field(..., min_length=1, max_length=50)
    start_date: datetime
    end_date: Optional[datetime] = None
    estimated_hours: Optional[float] = None
    budget: Optional[float] = None
    priority: Optional[Priority] = Priority.MEDIUM
    region: Optional[Region] = Region.BOTH
    project_lead_id: str


class UpdateProjectRequest(BaseModel):
    """Update project request schema"""
    name: Optional[str] = Field(None, min_length=1, max_length=200)
    description: Optional[str] = None
    client_name: Optional[str] = None
    status: Optional[ProjectStatus] = None
    start_date: Optional[datetime] = None
    end_date: Optional[datetime] = None
    estimated_hours: Optional[float] = None
    budget: Optional[float] = None
    priority: Optional[Priority] = None
    project_lead_id: Optional[str] = None


class UpdateProjectStatusRequest(BaseModel):
    """Update project status request schema"""
    status: ProjectStatus


class AddTeamMemberRequest(BaseModel):
    """Add team member request schema"""
    user_id: str
    role: str = Field(..., min_length=1, max_length=50)
    allocation_percentage: int = Field(100, ge=0, le=100)


class ProjectResponse(BaseModel):
    """Project response schema"""
    id: str
    name: str
    description: Optional[str] = None
    client_name: Optional[str] = None
    project_code: str
    status: str
    start_date: datetime
    end_date: Optional[datetime] = None
    estimated_hours: Optional[float] = None
    actual_hours: Optional[float] = None
    budget: Optional[float] = None
    priority: str
    region: str
    project_lead_id: str
    created_by_id: str
    created_at: datetime
    updated_at: datetime
    
    class Config:
        from_attributes = True

